testsuite/MDAnalysisTests/coordinates/base.py

```diff
@@ -92,3 +95,198 @@ def test_last_slice(self):
         trj_iter = self.universe.trajectory[-1:]
         frames = [ts.frame for ts in trj_iter]
         assert_equal(frames, np.arange(self.universe.trajectory.n_frames))
+
+
+class BaseReference(object):
+    def __init__(self):
+        self.trajectory = None
+        self.n_atoms = 5
+        self.n_frames = 5
+        # default for the numpy test functions
+        self.prec = 6
+        self.first_frame = np.arange(3 * self.n_atoms).reshape(self.n_atoms, 3)
+        self.second_frame = 2 ** 1 * self.first_frame
+        self.last_frame = 2 ** 4 * self.first_frame
+        self.jump_to_frame = 3  # second to last frame
+        # remember frames are 0 indexed
+        self.jump_to_frame_positions = 2 ** 3 * self.first_frame
+        self.has_forces = False
+        self.has_velocities = False
+        self.dimensions = np.array([80, 80, 80, 60, 60, 90], dtype=np.float32)
+        self.volume = mda.lib.mdamath.box_volume(self.dimensions)
+        self.time = 0
+        self.dt = 1
+        self.totaltime = 5
+
+
+class BaseReaderTest(object):
+    def __init__(self, reference):
+        self.ref = reference
+        self.reader = self.ref.reader(self.ref.trajectory)
+
+    def test_n_atoms(self):
+        assert_equal(self.reader.n_atoms, self.ref.n_atoms)
+
+    def test_n_frames(self):
+        assert_equal(len(self.reader), self.ref.n_frames)
+
+    def test_first_frame(self):
+        self.reader.rewind()
+        assert_array_almost_equal(self.reader.ts.positions,
+                                  self.ref.first_frame, decimal=self.ref.prec)
+
+    def test_double_close(self):
+        self.reader.close()
+        self.reader.close()
+        self.reader._reopen()
+
+    def test_reopen(self):
+        self.reader.close()
+        self.reader._reopen()
+        ts = self.reader.next()
+        assert_array_almost_equal(ts.positions, self.ref.first_frame,
```

Comment:
hm there is one problem with that. I need to be able to set a precision. XTC and PDB store their data in limited precision. Any good idea for that? I guess something like a general `MDAnalysis.coordinates.allclose' methods could be good. It would work like the normal equality operator of the Timestep but allow a tolerance in the values of position/velocity/forces.